### PR TITLE
WIP - allow catalog updateLocal defaults to ignore tip unless explicitly set and get only stable versions

### PR DESCRIPTION
**Pull request by zaheena**
https://github.com/rackn/product-backlog/issues/24

### CODE REVIEW

**Comment by galthaus** [comment]
Looks good.  The only question is does stable imply no tip.  

I think the version flag implies that version and newer.  Tip is newer.

**Comment by galthaus** [comment]
I suspect that what we really want is these patterns:
drpcli catalog updateLocal # gets all entries except tip
drpcli catalog updateLocal --version=v4.6.2  # All versions newer than v4.6.2 except tip
drpcli catalog uplodateLocal --version=stable # Stable only
drpcli catalog updateLocal --version=stable --tip # STsble + tip





8:11
Add --tip implies adding tip to all the others.
8:12
tip moves so much that this is probably reasonable.
8:12
We should proabably do this.  It is cleaner and defaults to smaller downloads.

File: cli/catalog.go [hunk 1]
```diff
@@ -374,15 +380,27 @@ and wind up in a file with the same name as the item + the default file extensio
...
-				if mv != nil {
-					if o, oerr := semver.NewVersion(v.ActualVersion); oerr != nil || mv.Compare(o) > 0 {
+				if minVersion == "" {
+					// We want to get everything here except tip (unless tip is set to true in which case we include tip)
+					if !tip && v.Tip {
+						continue
+					}
+				} else if requireStable {
+					// Only get things that are stable or tip if tip is set to true
+					if v.Version != "stable" || (tip && !v.Tip) {
```
Comment (galthaus) [inline cli/catalog.go:390]: It thinks this needs to be:

if !(v.Version == "stable" || (tip && v.Tip))

Which is also

if v.Version != "stable" && (!tip || !v.Tip) 

Victor prefers the first form.  I prefer the more readable form.  In this case, it is the first one, I think.

**Review by galthaus — CHANGES_REQUESTED** [review]
I think the logic is incorrect in the stable case.  Please review it.

File: cli/catalog.go (hunk 1 above)
Comment (zaheena) [reply cli/catalog.go]: You're right. I got lost a little here.
I think the third condition also needed to be fixed i.e only include tip if tip is set to true. Updated with a new commit.

**Pull request merged**